lib/src/main/java/com/diffplug/spotless/FeatureClassLoader.java

```diff
@@ -29,37 +31,31 @@
  * path of URLs.<br/>
  * Features shall be independent from build tools. Hence the class loader of the
  * underlying build tool is e.g. skipped during the the search for classes.<br/>
- * Only {@link #BUILD_TOOLS_PACKAGES } are explicitly looked up from the class loader of
- * the build tool and the provided URLs are ignored. This allows the feature to use
- * distinct functionality of the build tool.
+ *
+ * For `com.diffplug.spotless.glue.`, classes are redefined from within the lib jar
+ * but linked against the `Url[]`. This allows us to ship classfiles which function as glue
+ * code but delay linking/definition to runtime after the user has specified which version
+ * of the formatter they want.
+ *
+ *  For `"org.slf4j.` and (`com.diffplug.spotless.` but not `com.diffplug.spotless.extra.`)
+ * 	the classes are loaded from the buildToolClassLoader.
```

Comment:
@fvgh, just FYI, we finally have a way to ship glue code without requiring all deps of that glue code. It works like so:


- `com.diffplug.spotless.glue.` -> we define classes from the jars, which requires linking against `com.diffplug.spotless` from `buildToolClassLoader`
- we make an exception for `com.diffplug.spotless.extra`, which we always load from the jars as (we have in the past)
- we also load `org.slf4j.` from `buildToolClassLoader` (as we have in the past)

This will allow us to replace a lot of manual reflection code, alal #524.